chainer/link_hooks/spectral_normalization.py

```diff
@@ -240,14 +237,13 @@ def _prepare_parameters(self, link, input_variable=None):
             # Initialize the scaling parameter with the max singular value.
             weight_matrix = self.reshape_W(initialW.array)
             # TODO(crcrpar): Remove this when chainerx supports SVD.
-            if link.xp is chainerx:
-                xp, device, array = fallback._from_chx(weight_matrix)
-                if xp is numpy:
-                    _, s, _ = numpy.linalg.svd(array)
-                else:
-                    with chainer.using_device(device):
-                        _, s, _ = xp.linalg.svd(array)
-                s = fallback._to_chx(s)
+            device = link.device
+            if device.xp is chainerx:
+                fallback_device = device.fallback_device
+                weight_matrix_ = fallback_device.send(weight_matrix)
+                with chainer.using_device(fallback_device):
+                    _, s_, _ = fallback_device.xp.linalg.svd(weight_matrix_)
+                s = fallback_device.send(s_)
```

Comment:
I think this line should be `device` instead of `fallback_device`.
`tests/chainer_tests/link_hooks_tests/test_spectral_normalization.py`
failed when I verified it.

Reply:
It seems to be a mistake indeed, but which test was failing?
I can't replicate the failure.

Reply:
try it on the master branch :), probably you rebased before the chainerx for initializers patch was added.

Reply:
Thanks, replicated and fixed.